src/stc/gen_iface.py

```diff
@@ -1174,11 +1174,18 @@ def processIface(iface, h_tmplt, cpp_tmplt, ih_tmplt, h_dest, cpp_dest, docstr_d
     ih_text = ih_text % data
 
     # write out destination files
-    open(h_dest, 'w').write(h_text)
-    open(cpp_dest, 'w').write(cpp_text)
+    if sys.version_info[0] == 2:
+        acc = 'wb'
+        eol = {}
+    else:
+        acc = 'wt'
+        eol = {'newline':'\n'}
+
+    open(h_dest, acc, **eol).write(h_text)
+    open(cpp_dest, acc, **eol).write(cpp_text)
```

Comment:
Just to be sure: is the intention here to always write files using LF EOL under all platforms?

If so, why can't we just always use "wb", even for Python 3?

Sorry if I'm missing something obvious here again, but if using "wb" doesn't work, perhaps it could be briefly explained why it doesn't in a comment?

Reply:
Yes, this is to make sure LF EOL is used in the generated files (`stc.h`, `stc.cpp`). On Windows, at least for me, they got replaced with CRLF.

Python 3 requires `'newline':'\n'` to get the correct results. `wb` does not work:  `TypeError: a bytes-like object is required, not 'str'`. It will also work with only `w`, but for consistency I used `wt`.
Python 2 requires `wb` for correct results. It does not like `'newline':'\n'`: `TypeError: 'newline' is an invalid keyword argument for this function`. Using `wt` has no effect.

I'll add a comment to the code.